### DISCUSSION TITLE
Common point error handling mechanism (Audit Trail)

### DISCUSSION THREAD

**Discussion by shantanuneosoft** [NeoSOFT-Technologies/rest-node-nestjs#57] [Ideas]
We should have record of every API request from which IP its requesting.
If we blacklisted few IP's then it redirect to error handling functionality.

API Request -----> Middleware(where we can track the details of every API request) ---> Response

**Comment by suresh-shetiar** [comment]
If you see the list of discussion there is already one discussion started as `Common Error Response Handler`. If you are pointing to something else then please provide some more info. OR you can also add your points in the above mentioned discussion.
Thanks

**Reply by shantanuneosoft** [reply]
Yes, We can create common point middleware where we can create the module like audit trail. We should have record of every API request from which IP its requesting. 
If we blacklisted few IP's then it redirect to error handling functionality.

API Request -----> Middleware(where we can track the details of every API request) ---> Response
@suresh-shetiar @santoshshinde2012
